frontend/public/js/util.js

```diff
@@ -10,4 +10,31 @@
       function () { return $(this).data(prop) == val; }
     );
   };
-})(window.jQuery);
\ No newline at end of file
+})(window.jQuery);
+
+
+JSON.flatten = function(data) {
+    var result = {};
+    function recurse (cur, prop) {
+        if (Object(cur) !== cur) {
+            result[prop] = cur;
+        } else if (Array.isArray(cur)) {
+          /*
+             for(var i=0, l=cur.length; i<l; i++)
```

Comment:
i found code online to flatten JSON objects, which is how I decided to deal with nested objects in the metadata